### ISSUE TITLE
How to debug?

### ISSUE THREAD

**Issue by dav101** [google-ar/arcore-unity-sdk#39]
Guys how do you debug in Unity for arcore, I have a project I am making and the most I can do is build and run for a build on a Samsung 8 (wireless build) and then see logs in console, any way to connect to unity remote 5 or Visual Studio using Unity for arcore

**Comment by eisenbruch** [comment]
Build Settings > Enable Development Build and Script Debugging

Use Debug.Log() in your scripts.

**Comment by iQuentin** [comment]
@eisenbruch, i can't see any debug.log in Unity console despite the fact i've done all of your 3 tips mentionned above (plus selecting AndroidPlayer in Console -> Editor). 
Do you have any idea why ? 
Thanks.
(Details : Unity 2017.3 - Samsung S8 - apk built and run on the phone from Unity - ARcore project)

**Comment by eisenbruch** [comment]
@iQuentin when you have your app running and phone connected, are there any other options in the Console > Editor dropdown that you can connect? I think it only works if you have done a fresh Build & Run. It won't work if you just plug your phone in and run the app. See the youtube tutorials on ARCore for more info.

**Comment by iQuentin** [comment]
@eisenbruch, when running from a fresh B&R, here is what is can see in Console->Editor : 
- Player Logging   [checked]
- Editor
- AndroidPlayer(ADB@... )   [checked]
- <Enter IP>

Please note that it's the first time i'm trying to use this "remote debugging from app running on a phone back to Unity Console". Maybe i forgot something obvious.
(I'm currently searching youtube for ARCore debugging infos...)

**Comment by eisenbruch** [comment]
@iQuentin I am at work now, so I cannot help you at this time more than the following:
https://www.youtube.com/watch?v=u5Q6Io8IxfM
I think he goes over the debugging in that video, or maybe it is pt 1 of that tutorial.

**Comment by iQuentin** [comment]
Thanks for the link, @eisenbruch, unfortunately it didn't solve my problem. My current investigation is to check if the tcpip connection with the device is ok. For example the command "adb connect" doesn't answer anything good.
If, by chance, you know how the debug datas are sent back to the computer it could help me. I'm not familiar with with the "stack" or whatever it is. Is it through tcpip ? through a file somewhere ? through ADB ? or is it handled directly by Unity in some way ? 
As you see this will keep me up all night :-)

**Comment by dav101** [comment]
Hey man it’s really hard to debug in arcore but I managed to print to screen, make sure u are calling the adb in the correct path mins was in users/appdata/local/Android/sdk/platform tools

also I had to download a spectate platform tools and add that in this directory. If you find adb connect not working open up task manager and see how many adb tasks you have running could be that you have a adb server running and u run another one that’s why adb connect doesn’t connect. I had a problem like this so I used adb kill-server to kill the connect then tried adb devices saw my device typed adb tcpip 5555 connect to the server then adb IP address with 5555 as port. If your device refuses to connect could be firewall so disable that check you developer tools on the phone to make sure you have usb debugging setup and you are on the same network... try ping your phone as well. Also take out the connect wire when running and building in unity. I know it’s a lot but try and see if it helps...
 cheers Dav



On 1 Nov 2017, at 21:15, iQuentin <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks for the link, @eisenbruch<https://github.com/eisenbruch>, unfortunately it didn't solve my problem. My current investigation is to check if the tcpip connection with the device is ok. For example the command "adb connect" doesn't answer anything good.
If, by chance, you know how the debug datas are sent back to the computer it could help me. I'm not familiar with with the "stack" or whatever it is. Is it through tcpip ? through a file somewhere ? through ADB ? or is it handled directly by Unity in some way ?
As you see this will keep me up all night :-)

**Comment by iQuentin** [comment]
Thanks both for your advices, I'VE GOT IT !! 
Here is what I did, for someone else who may have the same problem : 
The hint was that the IP adress of the phone was external (10.61....) not local (192.168.....). I don't really know why but ADB could not connect to the device through the USB cable. Then I understood that the WiFi has to be ON (i switched it off before because a tutorial i found said that every other possible tcpip connection has to be turned off). The IP of the WiFi connection is of course local and this is the one i used to connect (with the command "adb connect IP_ADDRESS").

So, in a nutshell, THE SOLUTION IS : 
- be sure to have your WiFi connection of the device ON
- disconnect the USB cable (not sure it's necessary)
- use the local IP of the WiFi with the ADB command "connect"
- in Unity, of course, Console -> Editor -> AndroidPlayer@IP_ADDRESS

then i can see the debug.logs of my scripts in Unity Console while the ARcore app is running on the device.

**Comment by eisenbruch** [comment]
sounds like you have an adb issue. you should be able to use a cable for it. is your sdk/platform-tools fully updated?

**Comment by nathanmartz** [comment]
Closing as I don't think this is an ARCore specific issue.

**Comment by mainhoovishesh** [comment]
guys I'm not being able to build the app the pc is just making a sound, no notification or anything, just making a sound and not doing anything what can I do??

**Comment by tobeglad** [comment]
I can recommend you to use AR Foundation Editor Remote plugin.
I developed it while I was making an AR update for my game and decided to make it into separate plugin. Please read the Limitations sections. How it will be useful for someone.

AR Foundation Editor Remote plugin:
https://stampedegames.net/ar-foundation-remote.html
